docs/ObserveApiProposal.md

```diff
@@ -0,0 +1,782 @@
+#
+# This is simply an indication of what they contain.
+
+type GmosNorthStatic {
+
+  mosPreImaging: MosPreImaging!
+  
+  nodAndShuffle: NodAndShuffle
+  
+  stageMode:     GmosNorthStageMode!
+
+}
+
+type GmosNorthSequence {
+
+  # Atoms that make up the sequence, to be executed in order.
+  atoms: [GmosNorthAtom!]!
+
+  # Sum of all contained atoms' estimated times.
+  time:  StepTime!
+  
+}
+
+# Indivisible series of steps.  Breaking an atom implies losing the time
+# required to have obtained previous steps in the atom.  An example atom might
+# be a science frame and accompanying flat.  Without the flat, the science is
+# not usable.
+type GmosNorthAtom {
+
+  atomId: AtomId!
+  
+  # Sum of all step's estimated times.
+  time:  StepTime!
+  
+  # Steps that comprise the atom.  The must be executed in order.  
+  steps: [GmosNorthStep!]!
+
+}
+
+enum Breakpoint {
+
+  ENABLED
+  
+  DISABLED
+  
+}
+
+enum StepType {
+
+  BIAS
+  
+  DARK
+  
+  GCAL
+  
+  SCIENCE
+  
+  SMART_GCAL
+
+}
+
+type GmosNorthStep {
+
+  stepId:     StepId!
+
+  # Breakpoints that are programmed into the sequence generation, or present in
+  # manual sequence definitions.
+  breakpoint: Breakpoint!
+  
+  # The GMOS instrument configuration at this step, apart from step-specific
+  # details like offset positions or GCAL config.
+  instrument: GmosNorthDynamic!
+  
+  # Step config contains the details of bias, dark, gcal, science, etc. steps
+  # apart from the GMOS configuration itself, which is in `instrument`.  To
+  # do anything (other than find the `StepType`) we have to match on the
+  # particular StepConfig.  For example
+  #
+  # step {
+  #   ... on Gcal {
+  #     continuum
+  #     filter
+  #     shutter
+  #   }   
+  # }
+  step:       StepConfig!
+  
+  # Planned time estimate for this step
+  time:  StepTime!
+  
+}
+
+type GmosNorthDynamic {
+  exposure: Duration!
+  readout:  GmosCcdReadout!
+  dtax:     GmosDtax!
+  roi:      GmosRoi!
+  grating:  GmosNorthGrating
+  filter:   GmosNorthFilter
+  fpu:      GmosNorthFpu
+}
+
+interface StepConfig {
+ 
+  stepType: StepType!
+  
+}
+
+type Bias implements StepConfig {
+  stepType: StepType!
+}
+
+type Dark implements StepConfig {
+  stepType: StepType!
+}
+
+type Gcal implements StepConfig {
+  stepType:  StepType!
+  continuum: GcalContinuum
+  arcs:      [GcalArc!]!
+  filter:    GcalFilter!
+  diffuser:  GcalDiffuser!
+  shutter:   GcalShutter!
+}
+
+type Science implements StepConfig {
+  stepType:  StepType!
+  offset:    Offset!
+}
+
+```
+
+As discussed in the SDL comments above, it is expected that Observe will pull the next atom to
+execute at the beginning of the observation and each time thereafter that an atom is completed.
+This allows the course of the sequence to change over its execution according to conditions or
+other considerations. The `possibleFuture` shows what steps are anticipated for reference in the UI
+but should not be relied upon for actual execution.
```

Comment:
This will be a mayor change in transforming Seqexec into Observe. The internal model must reflect this `currentAtom` + `posibleFuture`. I guess the UI should also show the `possibleFuture` in distinctive way, and should give some clue to the user when the `possibleFuture`  is replaced by a new `currentAtom`

Reply:
Yes, that all sounds right to me.